ldms/src/ldmsd/test/ldmsd_stream_subscribe.c

```diff
@@ -175,6 +178,7 @@ static int stream_recv_ev(ldms_stream_event_t ev, void *arg)
 	if (!events_raw)
 		msglog("}");
 	msglog("\n");
+	MSGLOG_UNLOCK();
```

Comment:
I don't believe it has anything to do with the "scope of the lock" it has to do with the need to hold the lock across multiple calls to msglog(). This is probably what @morrone meant; so it's the description of the problem and the solution that was giving me pause. I was interpreting "scope" to mean "symbol scope."